### ISSUE TITLE
New linter: discourage the use of anonymous functions for trivial cases

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#1531]
# Example

## Discouraged

```r
lapply(l, function(x) mean(x)))
```

## Recommended

```r
lapply(l, mean)
```

# Scope

- to limit the risk of false positive, I'm thinking to limit this to the `apply()` family function (+ `replicate()`), at least as a first step. Or do you think it wouldn't be much more difficult and riskier to do this in all contexts?
- ideally, this should detect also both tidyverse (`~ mean(.x)`) and base (`\(x) mean(x)`) lambda functions
- I'm thinking to limit it to cases with a single argument, i.e. exclude:

  ```r
  lapply(l, function(x) mean(x, na.rm = TRUE))
  ``` 

  even though the anonymous function could be removed here as well by forwarding the `...` because users might find it clearer to not forward them.

**Comment by Bisaloo** [comment]
Can you see any potential issues / pitfalls?

If you agree this would be a good addition, I can start working on a PR soon (now that I understand a little better how the package works).

**Comment by MichaelChirico** [comment]
Hmm it seems I never commented here.

We already have such a linter internally, so it'll be covered by #884
